### PR TITLE
Load translated referenced entities before array_filter and abstract logic in a trait

### PR DESCRIPTION
**Pull request by drupov**
Currently the logic for resolving translated entity references is loading the translated versions in order to check the access to them. But at the end the default language version is returned. This PR makes sure to pass the proper language version(s) into the `array_filter` function that strips out the ones, where access is not given.

### CODE REVIEW

File: src/Plugin/GraphQL/DataProducer/Field/EntityReferenceTrait.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Drupal\graphql\Plugin\GraphQL\DataProducer\Field;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\TranslatableInterface;
+
+trait EntityReferenceTrait {
+
+  private function defer($type, $language, $bundles, $access, $accessUser, $accessOperation, $resolver, $context) {
```
Comment (fubhy) [inline src/Plugin/GraphQL/DataProducer/Field/EntityReferenceTrait.php]: I'd prefer this to be a protected method and also have a more descriptive name. By separating the function body of this function into smaller reusable chunks we'd also make it more customizable.

**Review by fubhy — CHANGES_REQUESTED** [review]

**Comment by fubhy** [comment]
Thanks for this PR Dimitar! Looks good in general but I'd like to split up the function body into smaller reusable pieces and use a more descriptive function name.

**Comment by drupov** [comment]
Abstracted also logic inside trait and added some comments to each method.

**Comment by chrfritsch** [comment]
The patch works nicely for me. One thing that is a bit related:
In the plugin definition language is defined as multiple. I think it shouldn't be.

**Comment by klausi** [comment]
This was merged in #1232 , thanks a lot for the upfront work here @drupov !

**Pull request closed without merging**